### PR TITLE
printing preview and resize

### PR DESCRIPTION
**Pull request by peggy-quartech**
# Description

This PR includes the following proposed change(s):

- spdbt-2450(BCMP/ OpenShift - Card Preview functionality)
- spdbt-2451(BCMP / OpenShift - Photo image resize functionality)

### CODE REVIEW

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        if (lics == null || !lics.Items.Any())
+        {
+            throw new ApiException(HttpStatusCode.BadRequest, "no licence found for the licenceId");
+        }
+        LicenceResp lic = lics.Items.First();
+        LicencePreviewJson preview = mapper.Map<LicencePreviewJson>(lic);
+        var serviceTypeListResp = await serviceTypeRepository.QueryAsync(
+                new ServiceTypeQry(null, Enum.Parse<ServiceTypeEnum>(preview.LicenceType)), cancellationToken);
+        string name = serviceTypeListResp.Items.First().ServiceTypeName;
+        preview.LicenceType = name;
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:44]: Please assign this value directly in previous line

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+    IIncidentRepository incidentRepository,
+    IMapper mapper)
+    : BcMailPlusTransformStrategyBase<PersonalLicencePreviewTransformRequest, LicencePreviewJson>(Jobs.SecurityWorkerLicense)
+{
+    protected override async Task<LicencePreviewJson> CreateDocument(PersonalLicencePreviewTransformRequest request, CancellationToken cancellationToken)
+    {
+        //following is for personal licence.
+        LicenceListResp lics = await licRepository.QueryAsync(new LicenceQry() { LicenceId = request.LicenceId }, cancellationToken);
+        if (lics == null || !lics.Items.Any())
+        {
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:36]: Please remove brakets as you did in line 53

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+        LicenceResp lic = lics.Items.First();
+        LicencePreviewJson preview = mapper.Map<LicencePreviewJson>(lic);
+        var serviceTypeListResp = await serviceTypeRepository.QueryAsync(
+                new ServiceTypeQry(null, Enum.Parse<ServiceTypeEnum>(preview.LicenceType)), cancellationToken);
+        string name = serviceTypeListResp.Items.First().ServiceTypeName;
+        preview.LicenceType = name;
+
+        LicenceApplicationResp app = await licAppRepository.GetLicenceApplicationAsync((Guid)lic.LicenceAppId, cancellationToken);
+        if (lic.WorkerLicenceTypeCode == WorkerLicenceTypeEnum.SecurityWorkerLicence)
+        {
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:48]: Please remove brackets as you did in line 53

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+        }
+        return names;
+    }
+
+    private async Task<string> EncodedPhoto(Guid documentUrlId, CancellationToken cancellationToken)
+    {
+        DocumentResp photoDoc = await documentRepository.GetAsync(
+            documentUrlId,
+            cancellationToken);
+        if (photoDoc == null) throw new ApiException(System.Net.HttpStatusCode.InternalServerError, "cannot find photograph for the applicant");
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:86]: Please add a line break for readability

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 5]
```diff
@@ -0,0 +1,247 @@
...
+    public string? Weight { get; set; } //67kg
+
+    [JsonPropertyName("temporaryLicence")]
+    public bool TemporaryLicence { get; set; }
+
+    [JsonPropertyName("cardType")]
+    public string? CardType { get; set; } //SECURITY-WORKER-AND-ARMOUR
+}
+
+public record BranchAddress
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:231]: Please add line breaks on this fields as you did in the previous ones

File: src/Spd.Resource.Repository/WorkerLicenceCategory/Contract.cs [hunk 6]
```diff
@@ -0,0 +1,24 @@
+using Spd.Resource.Repository.LicenceApplication;
+
+namespace Spd.Resource.Repository.WorkerLicenceCategory;
+public partial interface IWorkerLicenceCategoryRepository
+{
+    public Task<WorkerLicenceCategoryListResp> QueryAsync(WorkerLicenceCategoryQry query, CancellationToken cancellationToken);
+}
+
```
Comment (ruaraya) [inline src/Spd.Resource.Repository/WorkerLicenceCategory/Contract.cs:8]: Please remove extra line break

File: src/Spd.Utilities.Shared/Tools/StreamExtensions.cs [hunk 7]
```diff
@@ -0,0 +1,18 @@
+﻿namespace Spd.Utilities.Shared.Tools;
+public static class StreamExtensions
+{
+    public static string ConvertToBase64(this Stream stream)
+    {
+        if (stream is MemoryStream memoryStream)
+        {
```
Comment (ruaraya) [inline src/Spd.Utilities.Shared/Tools/StreamExtensions.cs:7]: Please remove extra brackets

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs [hunk 8]
```diff
@@ -0,0 +1,247 @@
...
+using Spd.Utilities.Shared.Tools;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
+using System.Net;
+using System.Text.Json.Serialization;
+
+namespace Spd.Manager.Printing.Documents.TransformationStrategies;
+
+internal class PersonalLicencePreviewTransformStrategy(ILicenceApplicationRepository licAppRepository,
```
Comment (ruaraya) [inline src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:21]: How is this class utilized?

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 1 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:44]: ok, fixed

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 3 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:48]: ok

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 2 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:36]: ok

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 4 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:86]: ok

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 5 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:231]: ?

File: src/Spd.Resource.Repository/WorkerLicenceCategory/Contract.cs (hunk 6 above)
Comment (peggy-quartech) [reply src/Spd.Resource.Repository/WorkerLicenceCategory/Contract.cs:8]: ok

File: src/Spd.Utilities.Shared/Tools/StreamExtensions.cs (hunk 7 above)
Comment (peggy-quartech) [reply src/Spd.Utilities.Shared/Tools/StreamExtensions.cs:7]: ok

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 8 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:21]: what do u mean?

Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:21]: It is used by dependence injection.

File: src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs (hunk 5 above)
Comment (peggy-quartech) [reply src/Spd.Manager.Printing/Documents/TransformationStrategies/PersonalLicencePreviewTransformStrategy.cs:228]: ok, fixed

**Review by ruaraya — APPROVED** [review]

**Pull request merged**